### PR TITLE
Designing the jobs page and making it responsive

### LINKED ISSUE #94: Jobs page - show the relevant jobs details to the user and add apply button

### PR DESCRIPTION
**Pull request by shakedch**
- Designing the page
- showing the job details when clicking on a relevant job
- adding message in case of “No Results"
- adding option to apply via url

Using: HTML, CSS, JavaScript

Fixes: #94 
@guy9050  @bardielle

image

image

### CODE REVIEW

**Comment by bardielle** [comment]
@shakedch please ask your team mates to review it before us

**Review by advapi — APPROVED** [review]

**Comment by advapi** [comment]
Looks great! @shakedch

**Review by mayasa3 — APPROVED** [review]

**Comment by shakedch** [comment]
> @shakedch please ask your team mates to review it before us

**Comment by shakedch** [comment]
Can you please review the PR?
@guy9050  @bardielle

File: searchJob/templates/job_posts/job_post_list.html [hunk 1]
```diff
@@ -28,31 +28,189 @@
...
+        for (var i = 0; i < classname.length; i++) {
+            classname[i].addEventListener('click', insertVariable);
+        }
+
+        function insertVariable(e) {
+            var name = this.getAttribute("data-name");
+            var company = this.getAttribute("data-company");
+            var city = this.getAttribute("data-city");
+            var type = this.getAttribute("data-type");
+			var scope = this.getAttribute("data-scope");
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html]: why do you need that indentation?

File: searchJob/templates/job_posts/job_post_list.html [hunk 2]
```diff
@@ -28,31 +28,189 @@
...
+			var description = this.getAttribute("data-description");
+			var url = this.getAttribute("data-url");
+            printVariables(name, company, city, type, scope, description, url);
+        }
+	
+	function printVariables(name, company, city, type, scope, description, url){
+		document.getElementById("job_details").style.display = "block";
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html]: why do you need that indentation?

File: searchJob/templates/job_posts/job_post_list.html (hunk 1 above)
Comment (shakedch) [reply searchJob/templates/job_posts/job_post_list.html]: For some reason when I open the code by IDE the indentation doesn't exist.. I will try to figure it out

Comment (shakedch) [reply searchJob/templates/job_posts/job_post_list.html]: Fixed

File: searchJob/templates/job_posts/job_post_list.html (hunk 2 above)
Comment (shakedch) [reply searchJob/templates/job_posts/job_post_list.html]: Fixed

File: searchJob/templates/job_posts/job_post_list.html [hunk 3]
```diff
@@ -28,36 +28,192 @@
...
+<title>joBoard</title> 
 
 </head>
 
 <body>
-  <h1> Available jobs  </h1>
 
-<div class="showSearch">
+    <div class="menu">
+		<button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton"
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html:47]: Fix the indentation

File: searchJob/templates/job_posts/job_post_list.html [hunk 4]
```diff
@@ -28,36 +28,192 @@
...
+          data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+          menu
+        </button>
+		<div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html:51]: Fix the indentation

File: searchJob/templates/job_posts/job_post_list.html [hunk 5]
```diff
@@ -28,36 +28,192 @@
...
+
+          <button onclick="location.href='{% url 'logout' %}'" class="btn btn-sm btn-outline-secondary"
+          type="button"><svg width="1em" height="1em" viewBox="0 0 16 16"
+          class="bi bi-person-plus-fill" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
+          <path fill-rule="evenodd" d="M1 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H1zm5-6a3 3 0 1 0 0-6 3 3 0 0 0
+           0 6zm7.5-3a.5.5 0 0 1 .5.5V7h1.5a.5.5 0 0 1 0 1H14v1.5a.5.5 0 0 1-1
+           0V8h-1.5a.5.5 0 0 1 0-1H13V5.5a.5.5 0 0 1 .5-.5z"/>
+          </svg> Log out </button>
+
+		{% else %}
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html:82]: Fix the indentation

File: searchJob/templates/job_posts/job_post_list.html [hunk 6]
```diff
@@ -28,36 +28,192 @@
...
+          <button onclick="location.href='{% url 'login' %}'" class="btn btn-sm btn-outline-secondary" type="button">
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html:83]: Fix the indentation from line 83 to line 98

File: searchJob/templates/job_posts/job_post_list.html [hunk 7]
```diff
@@ -28,36 +28,192 @@
...
+        {% endif %}
+    </div>
+    <h1> Available jobs  </h1>
+  
+  
 	{% if job_posts == "no relevant jobs for you" %}
-        <p>no relevant jobs for you<p>
+	<div class="no_job">
+        <p>Unfortunately there are no results matching your search..<br> Let’s try again</p>
+		<button class="search-btn" type="submit" onclick="history.back()" ><i class="fas fa-search"></i>
```
Comment (bardielle) [inline searchJob/templates/job_posts/job_post_list.html:115]: Fix the indentation (If Im not wrong need to like line 114)

Comment (shakedch) [reply searchJob/templates/job_posts/job_post_list.html:115]: I don't know why, but all the indentations look different in my local repo

for example:

PyCharm
image

notepad++

image

do you have any idea what is the reason? @bardielle

Comment (bardielle) [reply searchJob/templates/job_posts/job_post_list.html:115]: I have no idea :( 
maybe you have spaces instead of tabs?

**Pull request closed without merging**